### PR TITLE
Upgraded to rdfjs

### PR DESCRIPTION
**Pull request by LaurensRietveld**
See #35 

What changed:
- Upgraded to N3
- Rdfjs terms instead of strings for query arguments (e.g. the `searchTriples` function)
- Allow passing a datafactory as argument to `fromFile`
- Returning rdfjs terms from the functions. Done by looping over the results

What still needs to be done after this PR:
- Update README
- Update type definitions

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
Thanks! Do you happen to have any insights in performance regressions?

**Comment by RubenVerborgh** [comment]
Pinging @mielvds and @rubensworks for a review.

Note: is semver major.

**Comment by LaurensRietveld** [comment]
Didn't do any testing yet (other than the test suite), so can't say much about the performance. Good thing is that the majority of the work in this pr (the test suite) is useful regardless of the solution (c vs js) we end up with.

File: lib/hdt.js [hunk 1]
```diff
@@ -64,21 +71,21 @@ HdtDocumentPrototype.searchTerms = function (options) {
...
   }
-
+  const dataFactory = this.dataFactory;
   // Return distinct terms
   return new Promise((resolve, reject) => {
     if ('subject' in options || 'object' in options) {
       if (!subject && !object) return resolve([]);
-      this._fetchDistinctTerms(subject, object, limit, posId,
-        (error, results) => error ? reject(error) : resolve(results));
+      this._fetchDistinctTerms(termToString(subject) || '', termToString(object) || '', limit, posId,
```
Comment (mielvds) [inline lib/hdt.js]: termToString is guarenteed to return null or undefined if it cannot parse? @rubensworks ?

Comment (rubensworks) [reply lib/hdt.js]: It will return null if the term is falsy: https://github.com/rubensworks/rdf-string.js/blob/master/lib/TermUtil.ts#L23-L25

Comment (mielvds) [reply lib/hdt.js]: Then we are good to go!

**Comment by RubenVerborgh** [comment]
Thanks a lot @LaurensRietveld. Leaving #45 to do some checks before we release.

**Comment by mielvds** [comment]
And update the docs...

**Comment by RubenVerborgh** [comment]
@mielvds Yes, thx: https://github.com/RubenVerborgh/HDT-Node/issues/46

**Pull request merged**